libs/designer-ui/src/lib/tokenpicker/tokenpickersection/tokenpickerhelpers.ts

```diff
@@ -6,14 +6,15 @@ export const getReducedTokenList = (
 ): OutputToken[] => {
   const { hasSearchQuery, maxRowsShown, showAllOptions } = options;
 
-  let filteredTokens = tokens.filter((token) => !token.isAdvanced || showAllOptions || hasSearchQuery);
+  // Only filter if there are more tokens than maxRowsShown
+  let filteredTokens = tokens;
+  if (tokens.length > maxRowsShown) {
```

Comment:
I think this will produce some inconsistent behavior. We should create some more unit test cases and confirm it.

These are the cases I'm considering (with max = 6):

1. 0 non-advanced tokens, 1 advanced token
2. 0 non-advanced tokens, 5 advanced tokens
3. 0 non-advanced tokens, 6 advanced tokens
4. 0 non-advanced tokens, 7 advanced tokens
5. 5 non-advanced tokens, 0 advanced tokens
6. 6 non-advanced tokens, 0 advanced tokens
7. 7 non-advanced tokens, 0 advanced tokens
8. 5 non-advanced tokens, 1 advanced token
9. 6 non-advanced tokens, 1 advanced token
10. 5 non-advanced tokens, 2 advanced tokens

Let's make sure we've defined what is shown in each of these cases, and make sure we have unit tests covering each.

Reply:
Pushed changes to tokenpickerhelper and added the tests
![image](https://github.com/user-attachments/assets/28039e75-9b3f-4aa1-bd2f-8db03052d4ca)